api/library/lua-apclientpp.lua

```diff
@@ -0,0 +1,342 @@
+function APClient:set_room_info_handler(callback) end
+
+---Callback will be called when ConnectSlot succeeded.
+---@param callback fun(slot_data:{[string]: any}):nil
+function APClient:set_slot_connected_handler(callback) end
+
+---Callback will be called when ConnectSlot failed.
+---@param callback fun(reasons:string[]):nil
+function APClient:set_slot_refused_handler(callback) end
+
+---Callback will be called when items were received. Use item.index to see which are new.
+---@param callback fun(items:NetworkItem[]):nil
+function APClient:set_items_received_handler(callback) end
+
+---Callback will be called when locations were scouted.
+---@param callback fun(items:NetworkItem[]):nil
+function APClient:set_location_info_handler(callback) end
+
+---Callback will be called when locations were checked (e.g. a different player connected to the same slot).
+---@param callback fun(locations:integer[]):nil
+function APClient:set_location_checked_handler(callback) end
+
+---Callback will be called when data package changed (was updated). Use is_data_package_valid to see if this was the final update.
+---@param callback fun(data_package:{[string]: any}):nil
+function APClient:set_data_package_changed_handler(callback) end
+
+---Callback will be called when a legacy print message was received. This should not happen anymore.
+---@param callback fun(message:string):nil
+function APClient:set_print_handler(callback) end
+
+---Callback will be called when a PrintJSON message was received.
+---Use `render_json(data)` to get a human readable version.
+---`command` is the complete command and can be used to filter messages.
+---@param callback fun(data:{[string]: any}, command:{[string]: any}):nil
+function APClient:set_print_json_handler(callback) end
+
+---Callback will be called when a Bounced message was received.
+---@param callback fun(command:{[string]: any}):nil
+function APClient:set_bounced_handler(callback) end
+
+---Callback will be called as response to Get.
+---`data` is a table key -> value.
+---`keys` is the list of keys. This is required because a `nil` value can't be checked for existence in data.
+---`command` is the raw command including `extra`.
+---@param callback fun(data:{[string]: any}, keys:string[], command:{[string]: any})
+function APClient:set_retrieved_handler(callback) end
+
+---Callback will be called as response to Set.
+---@param callback fun(command:{[string]:any}):nil
+function APClient:set_set_reply_handler(callback) end
+
+
+-- Commands --
+
+---Send chat message.
+---@param text string
+---@return boolean true if message was queued
+function APClient:Say(text) end
+
+---Connect to a slot.
+---@param name string name of the slot to connect
+---@param password string
+---@param items_handling ItemsHandling describes which items to receive from the server
+---@param tags string[]? optional list of tags to use, e.g. `{"DeathLink"}`
+---@param version integer[]? optional client version, e.g. `{0, 5, 0}`
```

Comment:
do we want to explain the verison format here?
```suggestion
---@param version integer[]? optional client version in the format of `{major, minor, build}`, e.g. `{0, 5, 0}`
```